src/todo.js

```diff
@@ -0,0 +1,143 @@
+function listItm(todos) {
+  const itemlist = document.createElement('li');
+  const checkbox = document.createElement('input');
+  checkbox.className = 'check';
+
+  checkbox.addEventListener('change', () => {
+    if (checkbox.nextElementSibling.style.textDecoration !== 'line-through') {
+      checkbox.nextElementSibling.style.textDecoration = 'line-through';
+    } else {
+      checkbox.nextElementSibling.style.textDecoration = 'none';
+    }
+    const data = JSON.parse(localStorage.getItem('data'));
+    const indx = todos.index;
+    const action = data.filter((todo) => indx === todo.index);
+    if (action[0].completed === false) {
+      action[0].completed = true;
+      data.splice(indx - 1, 1, action[0]);
+      localStorage.setItem('data', JSON.stringify(data));
+    } else {
+      action[0].completed = false;
+      data.splice(indx - 1, 1, action[0]);
+      localStorage.setItem('data', JSON.stringify(data));
+    }
+  });
+  checkbox.setAttribute('type', 'checkbox');
+  checkbox.checked = todos.completed;
+  const todoField = document.createElement('input');
+  todoField.className = 'todo-edt-input';
+  todoField.setAttribute('type', 'text');
+  todoField.addEventListener('input', () => {
+    if (todoField.value !== '') {
+      const data = JSON.parse(localStorage.getItem('data'));
+      const indx = todos.index;
+      const action = data.filter((todo) => indx === todo.index);
+      action[0].description = todoField.value;
+      data.splice(indx - 1, 1, action[0]);
+      localStorage.setItem('data', JSON.stringify(data));
+    }
+  });
+  const iconSpan = document.createElement('span');
+  iconSpan.className = 'symbol-option-cont';
+  iconSpan.innerHTML = '&#128465;';
+  iconSpan.addEventListener('click', () => {
+    const data = JSON.parse(localStorage.getItem('data'));
```

Comment:
According to the DRY principle, it will be nice if you can have a function to perform this task